src/universal/Atmosphere.js

```diff
@@ -32,31 +70,60 @@ export default class Atmosphere extends Environment {
 
     // now atmosphere
     this.authToken = undefined;
-    this.socket = undefined;
+    this.subscriptionClient = undefined;
     this.networks = {
       http: this._network,
       socket: Network.create(this.fetchWS, this.socketSubscribe)
     };
-
-    this.opIdIndex = 1;
     this.querySubscriptions = [];
     this.subscriptions = {};
+    this.useSubscriptions = true;
   }
 
-  use = (networkName) => {
-    // TODO build this out when we start hitting github directly
-    return this.environments[networkName];
-  };
+  makeSubscriptionClient() {
+    if (!this.subscriptionClient) {
+      this.subscriptionClient = new Promise((resolve, reject) => {
+        const subscriptionClient = new SubscriptionClient(`ws://${window.location.host}/?token=${this.authToken}`, {
+          reconnect: true
+        });
+        // Catch aggressive firewalls that block websockets
+        subscriptionClient.client.onerror = (e) => {
+          // if reconnecting, then websockets aren't blocked
+          if (subscriptionClient.reconnecting) return;
+          subscriptionClient.reconnect = false;
+          reject(e);
+        };
 
-  addNet = (name, network) => {
-    this.networks[name] = network;
-    this.setNet(name);
+        subscriptionClient.client.onopen = () => {
+          resolve(subscriptionClient);
+          subscriptionClient.clearMaxConnectTimeout();
+          subscriptionClient.closedByUser = false;
+          subscriptionClient.eventEmitter.emit(subscriptionClient.reconnecting ? 'reconnecting' : 'connecting');
+          subscriptionClient.flushUnsentMessagesQueue();
+        };
+      });
+    }
+    return this.subscriptionClient;
+  }
+
+  fetchWS = async (operation, variables) => {
+    const request = this.subscriptionClient
+      .request({query: operation.text, variables});
+    return new Promise((resolve, reject) => {
+      request.subscribe({
+        // next: reject,
+        // next: ((data) => {
+        //   setTimeout(() => resolve(data), 1000)
+        // }),
```

Comment:
Can delete

Reply:
nah, i use this every time we I test a new error & loading state. super handy!